src/event_loop.rs

```diff
@@ -378,6 +262,72 @@ impl<T> EventLoopWindowTarget<T> {
         #[cfg(any(x11_platform, wasm_platform, wayland_platform, windows))]
         self.p.listen_device_events(_allowed);
     }
+
+    /// When the current loop iteration finishes, immediately begin a new iteration regardless of
+    /// whether or not new events are available to process.
+    ///
+    /// This is the default.
+    pub fn set_poll(&self) {
+        self.p.set_poll()
+    }
+
+    /// When the current loop iteration finishes, suspend the thread until another event arrives.
+    pub fn set_wait(&self) {
+        self.p.set_wait()
+    }
+
+    /// When the current loop iteration finishes, suspend the thread until either another event
+    /// arrives or the given time is reached.
+    ///
+    /// Useful for implementing efficient timers. Applications which want to render at the display's
+    /// native refresh rate should instead use [`set_poll()`] and the VSync functionality of a graphics API
+    /// to reduce odds of missed frames.
+    ///
+    /// [`set_poll()`]: Self::set_poll
+    pub fn set_wait_until(&self, instant: Instant) {
+        self.p.set_wait_until(instant)
+    }
+
+    /// Sets this to wait until a timeout has expired.
+    ///
+    /// In most cases, this is uses [`set_wait_until()`]. However, if the timeout overflows, it
+    /// instead uses [`set_wait()`].
+    ///
+    /// [`set_wait_until()`]: Self::set_wait_until
+    /// [`set_wait()`]: Self::set_wait
+    pub fn set_wait_timeout(&self, timeout: Duration) {
+        match Instant::now().checked_add(timeout) {
+            Some(instant) => self.set_wait_until(instant),
+            None => self.set_wait(),
+        }
+    }
+
+    /// Send a [`LoopExiting`] event and stop the event loop. This variant is *sticky* - once set,
+    /// it cannot be changed, and any future attempts to do so will
+    /// result in it being reset to it's previous state.
+    ///
+    /// The contained number will be used as exit code. The [`set_exit()`] method is a shortcut for this
+    /// with exit code 0.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **Android / iOS / Web:** The supplied exit code is unused.
+    /// - **Unix:** On most Unix-like platforms, only the 8 least significant bits will be used,
+    ///   which can cause surprises with negative exit values (`-42` would end up as `214`). See
+    ///   [`std::process::exit`].
+    ///
+    /// [`LoopExiting`]: Event::LoopExiting
+    /// [`set_exit()`]: Self::set_exit
+    pub fn set_exit_with_code(&self, code: i32) {
```

Comment:
Now that `EventLoop::run` returns (on most platforms), do we actually need this? I get that it's convenient, but it's not really cross-platform, and applications that want to return a proper status code can just do so.

Reply:
Yeah, now that I think about it the whole "exit code" thing might not be necessary now that `EventLoop` isn't the root loop of the application.

Reply:
Agreed, right now it actually just returns an error code in `EventLoopError::ExitFailure(code)`, that sounds pretty useless to me.

Reply:
Done.